src/include/daos/mgmt.h

```diff
@@ -53,7 +53,8 @@ struct dc_mgmt_sys {
 	struct dc_mgmt_sys_info	sy_info;
 };
 
-int dc_mgmt_sys_attach(const char *name, struct dc_mgmt_sys **sysp);
+int
+	dc_mgmt_sys_attach(const char *name, bool use_global_info, struct dc_mgmt_sys **sysp);
```

Comment:
im not quite sure i understand you question, so please clarify if my answer does not answer it.
we use the cached info if use_global_info is passed as true (basically in net_cfg, proto query, and pool connect calls).

Reply:
Oh sorry. To clarify, I'm just wondering why we would have some code paths use the locally cached version and some not. Why not have them all use the cache? Or have some env variable to make it configurable, if you envision cases where we may not want it cached?

Reply:
there are some cases where i am not familiar with the code that i chose to keep the default behavior (the server side attach that uses the client stack and the rbt cases). if someone more familiar with the code thinks otherwise, we can update those too and remove this flag. I was just trying to address the ticket where just a simple daos command is triggering 5-6 attach calls to the agent.

Reply:
Ah, understood. From my POV I don't understand why any individual client process would need to query the agent multiple times for the same info. The only element that could change is the suggested interface/domain.

Reply:
well what im saying that it's not the client only that's using that function. see:
pool/srv_cli.c
rdb/tests/rdbt.c

for the former i think the flag does not matter because we do not call the agent attach anyway since this is a server process. but for the other rdbt test, i do not know. So i think with some review from code owners, this flag might be removed, but for now i added to maintain the original behavior in those cases.

Reply:
@kccain Any thoughts about the rdbt test?

Reply:
Thanks.